seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/BaseFileSinkWriter.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.file.sink;
+
+import org.apache.seatunnel.api.sink.SinkWriter;
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+import org.apache.seatunnel.connectors.seatunnel.file.config.HadoopConf;
+import org.apache.seatunnel.connectors.seatunnel.file.sink.commit.FileCommitInfo2;
+import org.apache.seatunnel.connectors.seatunnel.file.sink.state.FileSinkState2;
+import org.apache.seatunnel.connectors.seatunnel.file.sink.writer.WriteStrategy;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+public class BaseFileSinkWriter implements SinkWriter<SeaTunnelRow, FileCommitInfo2, FileSinkState2> {
+    private final WriteStrategy writeStrategy;
+    private final HadoopConf hadoopConf;
+    private final SinkWriter.Context context;
+    private final int subTaskIndex;
+    private final String jobId;
+
+    public BaseFileSinkWriter(WriteStrategy writeStrategy, HadoopConf hadoopConf, SinkWriter.Context context, String jobId) {
+        this.writeStrategy = writeStrategy;
+        this.context = context;
+        this.hadoopConf = hadoopConf;
+        this.jobId = jobId;
+        this.subTaskIndex = context.getIndexOfSubtask();
+        writeStrategy.init(hadoopConf, jobId, subTaskIndex);
```

Comment:
Use `this(writeStrategy, hadoopConf, context, jobId, Collections.EMPTY_LIST)` ?

Reply:
Good advice.